### PR TITLE
[CEED] Add Full Matrix Free operators using libCEED

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR adds full matrix free operators using libCEED.

Full Matrix Free Mass and Diffusion operators can be used with the `-mf` option in ex1.

### CODE REVIEW

**Comment by artv3** [comment]
@YohannDudouit , what is meant by full matrix free?

File: fem/bilininteg.hpp [hunk 1]
```diff
@@ -1678,6 +1678,8 @@ class DiffusionIntegrator: public BilinearFormIntegrator
    const GeometricFactors *geom;  ///< Not owned
    int dim, ne, dofs1D, quad1D;
    Vector pa_data;
+   // CEED extension
+   void* ceedDataPtr;
```
Comment (artv3) [inline fem/bilininteg.hpp]: Will this make libceed a required mfem dependency?

Comment (YohannDudouit) [reply fem/bilininteg.hpp]: Absolutely nothing is stored or precomputed, everything is computed on the fly.

**Comment by YohannDudouit** [comment]
@artv3 Absolutely nothing is stored or precomputed, everything is computed on the fly.

File: fem/bilininteg.hpp (hunk 1 above)
Comment (YohannDudouit) [reply fem/bilininteg.hpp]: No, you have to compile with `MFEM_USE_CEED=YES` to use libCEED. I could guard this `void*` with `#ifdef MFEM_USE_CEED`, but a `void*` is not a dependency issue by itself.

**Comment by artv3** [comment]
> @artv3 Absolutely nothing is stored or precomputed, everything is computed on the fly.

Neat, looking forward to trying it out.

**Comment by YohannDudouit** [comment]
@artv3 It is unfortunately currently slower than Partial Assembly. However, if your mesh is moving   and you're only using the operator once, then Full Matrix Free is faster.

**Comment by tzanio** [comment]
Does this depend on #866

**Comment by YohannDudouit** [comment]
In some way it even depends on #1100 (as it includes it)

File: general/device.hpp [hunk 2]
```diff
@@ -236,7 +236,7 @@ class Device
    /** @brief Get the current Device MemoryClass. This is the MemoryClass used
        by most MFEM device kernels to access Memory objects. */
    static inline MemoryClass GetMemoryClass() { return Get().mem_class; }
-   
+
    static void SetGPUAwareMPI(const bool force = true)
    { Get().mpi_gpu_aware = force; }
```
Comment (artv3) [inline general/device.hpp]: @camierjs , I didn't know we had GPU Aware MPI! Is this new? Has it shown performance improvements?

**Comment by YohannDudouit** [comment]
Closing this in favor of #1670

**Pull request closed without merging**